service/history/resource/resourceTest.go

```diff
@@ -26,13 +26,15 @@ import (
 	"github.com/uber/cadence/common/metrics"
 	"github.com/uber/cadence/common/resource"
 	"github.com/uber/cadence/service/history/events"
+	"github.com/uber/cadence/service/history/failover"
 )
 
 type (
 	// Test is the test implementation used for testing
 	Test struct {
 		*resource.Test
-		EventCache *events.MockCache
+		eventCache  *events.MockCache
+		coordinator *failover.MockCoordinator
```

Comment:
Please export those two fields. It will make test code easier to write, otherwise you will need to call `GetXXX()` and do a type assertion every time.

Reply:
updated